ui/packages/catalog-ui-search/src/main/webapp/component/select/filterHelper.spec.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import { expect } from 'chai'
+
+var filterHelper = require ('./filterHelper.js')
+
+describe('fitler attrs', () => {  
+    it('filter anyText', () => {
+      var str = "anyText"
+      var matchcase = false
+
+      var filterValue = "w"
+      expect(filterHelper.matchesFilter(filterValue, str, matchcase)).to.equal(false)
```

Comment:
I would combine some of these test cases into a few big data structure to loop through and reduce duplication.

Reply:
That's what I was thinking too, just wanted to get something out there 👍